### PR TITLE
CatLog Of My Things

### PR DESCRIPTION
**Pull request by MudasirSherwani**
## Hi Sir, 👋 
Our team consists of 3 members 👍 
**@george827 , @MudasirSherwani  and @admirerbrown** 
## Code Work:
- [ ] Each class is created in a separate .rb file.
- [ ] All associations implemented.
- [ ] All methods visible in the diagram should be implemented:
- [ ] add_item method (in all classes that have that method)
- [ ] can_be_archived?() in the Item class
- [ ] move_to_archive() in the Item class
- [ ] can_be_archived?() in the Book class
- [ ] can_be_archived?() in the MusicAlbum class
- [ ] can_be_archived?() in the Movie class
- [ ] can_be_archived?() in the Game class
- [ ] Add unit tests for all implemented methods.
- [ ] All data reserved in JSON files
- [ ] Schema of SQL database is created.

## Interface Includes: 
1. List all books
2. List all music albums
3. List of games
4. List all genres
5. List all labels
6. List all authors
7. Add a book
8. Add a music album
9. Add a game
10. Have a way to quit the app.

## Other Requirements:
- [ ] Add Descriptive readme file
- [ ] Add video documentation 
- [ ] Use GitFlow
- [ ] Fix Linters
- [ ]

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- GETTING STARTED -->
+## demo-video
+> group demo video for the presentation of the app
+
+[Link to demo video](https://youtu.be/qYq3NXRCJrU)
```
Comment (ShadyShawkat) [inline Readme.md]: - [x] Good job recording a video. However, 11 minutes is way longer than the required length of the video. Please make sure that the video has a maximum length of 5 minutes. As per [this](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/catalog_of_my_things.md#:~:text=It%20is%20a%20video%20with%20a%20maximum%20length%20of%205%20minutes.) requirement.

File: schema/schema.sql [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  publish_date DATE,
+  archived BOOLEAN,
+  on_spotify BOOLEAN,
+  item_id INT NOT NULL,
+
+  CONSTRAINT fk_item
+      FOREIGN KEY (item_id)
+      REFERENCES item (id)
+      ON DELETE CASCADE
+);
```
Comment (ShadyShawkat) [inline schema/schema.sql]: - [x] Kindly create indexes for each of the created foreign keys

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sumairq — APPROVED** [review]
### Hi Team, 



In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**